src/ErrorHandler.php

```diff
@@ -111,52 +100,72 @@ private function __construct(callable $callback, int $reservedMemorySize = self:
             // first call to the set_error_handler method would cause the PHP
             // bug https://bugs.php.net/63206 if the handler is not the first
             // one in the chain of handlers
-            set_error_handler([$this, 'handleError'], $this->capturedErrors);
-
-            $this->isRoot = true;
+            set_error_handler([$this, 'handleError'], E_ALL);
         }
 
         $this->previousExceptionHandler = set_exception_handler([$this, 'handleException']);
     }
 
     /**
-     * Registers this error handler and associates the given callback to the
-     * instance.
-     *
-     * @param callable $callback           callback that will be called when exception is caught
-     * @param int      $reservedMemorySize The amount of memory to reserve for the fatal error handler
+     * Gets the current registered error handler; if none is present, it will register it.
      *
-     * @return self
+     * @param int|null $reservedMemorySize The requested amount of memory to reserve
      *
-     * @throws \ReflectionException
+     * @return ErrorHandler The ErrorHandler singleton
      */
-    public static function register(callable $callback, int $reservedMemorySize = self::DEFAULT_RESERVED_MEMORY_SIZE): self
+    public static function getRegisteredHandler(int $reservedMemorySize = null): self
     {
-        return new self($callback, $reservedMemorySize);
+        if (null === self::$registeredHandler) {
+            self::$registeredHandler = new self($reservedMemorySize ?? self::DEFAULT_RESERVED_MEMORY_SIZE);
+        } elseif (null !== $reservedMemorySize) {
```

Comment:
It's to avoid inconsistency. 

I need to expose the parameter at least for the first registration, so I cannot ignore it if the user calls it more than once. The user could not know that the error handler was registered before.

Reply:
Renaming this method to something else like `register` or `registerOnce` along with a DocBlock description could help the user better understand what this method does and which rules it follows (e.g. it registers a singleton, so multiple calls have no effect). I fear that leaving open the possibility of changing after initialization the amount of reserved memory could have some implications that we don't know of and also I don't see any valid reason someone should do it

Reply:
No issues whatsoever, the effect would be the same as registering once as with that memory. Only point where that changes is when setting to `0` while handling a fatal error, but that's a point of no return, PHP is crashing, so the user cannot make any mess anyway.

Reply:
One more thing that doesn't convince me at all is that since this is a getter I don't expect it to edit the object that it returns

Reply:
Well, it's already a magic getter since it triggers the singleton instantiation. Maybe we can rename it?

Reply:
But triggering the singleton instantiation is different from being able to edit it on each subsequent call to the method, isn't it?

Reply:
I agree with @ste93cry rename it to registerOnce and remove the setter. That way it make the most sense to me.

Reply:
Done. I had to move a test to a PHPT.